open-craft/microsite-generator
# [BB-4023] Create MicroSite Configuration on different services

## Description

This PR adds a set of scripts to generate microsite-related configurations for `LMS`, `discovery`, and `ecommerce` services from a given configuration file.


## Supporting information

https://tasks.opencraft.com/browse/BB-4023

## Testing instructions

1. Pull this PR to `src` directory of your local devstack
2. Copy `config/sample.yaml` to `config/config.yaml`
3. To generate & test LMS configurations -
    - Login to LMS shell - `make lms-shell`
    - Go to microsite directory - `cd /edx/src/microsite-generator`
    - Run - `make dev.run-lms`
    - Check if `Site`, `SiteConfigurations`, `Organizations` has been created from LMS admin.
4. To generate & test Discovery configurations -
    - Login to Discovery shell - `make discovery-shell`
    - Go to microsite directory - `cd /edx/src/microsite-generator`
    - Run - `make dev.run-discovery`
    - Check if `Site`, `Partner` has been created from Discovery admin.
5. To generate & test eCommerce configurations -
    - Login to eCommerce shell - `make ecommerce-shell`
    - Go to microsite directory - `cd /edx/src/microsite-generator`
    - Run - `make dev.run-ecommerce`
    - Check if `Site`, `Partner`, `SiteConfiguration` has been created from eCommerce admin.
 6. Check features in README.md of this PR, and play with the `config/config.yaml` to test other features.
  

## Deadline

ASAP

## Other information
